File: src/Systems/Swift/patterns/Visitor.swift
enum VisitorExample {
    enum Shape {
        case circle(Double)
        case rect(Double, Double)
    }

    static func area(_ shape: Shape) -> Double {
        switch shape {
        case .circle(let radius): return Double.pi * radius * radius
        case .rect(let width, let height): return width * height
        }
    }

    static func run() -> Bool {
        let total = [Shape.circle(2), .rect(3, 4)].map(area).reduce(0, +)
        return abs(total - (4 * Double.pi + 12)) < 1e-9
    }
}

File: src/Systems/Fortran/patterns/visitor.f90
module visitor_example
  implicit none
contains
  logical function run()
    real(8) :: total, expected
    total = circle_area(2.0d0) + rectangle_area(3.0d0, 4.0d0)
    expected = 4.0d0 * acos(-1.0d0) + 12.0d0
    run = abs(total - expected) < 1.0d-9
  contains
    real(8) function circle_area(radius)
      real(8), intent(in) :: radius
      circle_area = acos(-1.0d0) * radius * radius
    end function
    real(8) function rectangle_area(width, height)
      real(8), intent(in) :: width, height
      rectangle_area = width * height
    end function
  end function
end module
